### ISSUE TITLE
sized_box_for_whitespace has false negatives

### ISSUE THREAD

**Issue by gspencergoog** [dart-lang/linter#2494]
**Describe the issue**

If sized_box_for_whitespace is enabled, it doesn't flag cases like this:

```
Container(height: 100.0),
Container(width: 100.0),
Container(),
```

If both width and height are specified, then it does flag them.

**Expected behavior**
I expected the above to be flagged.

**Comment by a14n** [comment]
Without `child` parameter the Container [adds a `LimitedBox`](https://github.com/flutter/flutter/blob/77925192aaa6d2725ea6c194f9aa7218d9906095/packages/flutter/lib/src/widgets/container.dart#L386-L392) in the widget tree. Replacing `Container` by `SizedBox` may change the rendering in some cases. That why we don't lint such cases (to limit false positives).

**Comment by gspencergoog** [comment]
Well, but it gives the warning for `Container(width: 100.0, height: 100.0)`.  The implementation appears to also add a limited box in those cases as well, so should `Container(height: 100.0)` also warn, or should `Container(width: 100.0, height: 100.0)` not warn?

**Comment by a14n** [comment]
Actually no `LimitedBox` is added if `width` and `height` are both provided because they are used to create a tightened `constraints`.

**Comment by gspencergoog** [comment]
OK, I guess it's correct then, if confusing.
